File: codes/medium/55 Jump Game/discussion.c
bool canJump(int* nums, int numsSize){
    int last = numsSize - 1;

    for(int i = numsSize - 1; i >= 0; --i) {
        if(nums[i] + i >= last) last = i;
    }

    return last == 0;
}

File: codes/medium/55 Jump Game/discussion.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        last = n - 1

        for i in range(n - 1, -1, -1):
            if(nums[i] + i >= last):
                last = i
        
        return last == 0